src/module.c

```diff
@@ -3150,16 +3281,9 @@ int RSSearchCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc);
 static int DistSearchCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   if (NumShards == 0) {
```

Comment:
Not necessarily in this PR - consider using `!SearchCluster_Ready`

Reply:
(also in 2 more places)

Reply:
Agree, let's do it separately